File: Practica01/code/SelectionSort/SelectionSort.java
import java.util.Scanner;

public class SelectionSort {
	static void sort(int A[]) {
		int n = A.length;
		for (int i = 0; i < n - 1; i++) {
			int menor = i;
			for (int j = i + 1; j < n; j++)
				if (A[j] < A[menor])
					// Encontrar el N°menor
					menor = j;
			// intercambiamos con la posicion i el N°menor
			int temp = A[menor];
			A[menor] = A[i];
			A[i] = temp;
		}
	}

	static void print(int[] a) {
		for (int i = 0; i < a.length; ++i) {
			System.out.print(a[i] + " ");
		}
		System.out.println();
	}

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		while (sc.hasNextInt()) {
			int n = sc.nextInt();
			int[] arr = new int[n];
			for (int i = 0; i < n; ++i) {
				int x = sc.nextInt();
				arr[i] = x;
			}
			long start = System.nanoTime();
			sort(arr);
			long end = System.nanoTime();
			// print(arr);
			System.out.print((end - start) / 1000000000.0 + " ");
		}
		System.out.println();
		sc.close();
	}

}

File: Practica01/code/SelectionSort/SelectionSort.cpp
#include <iostream>
#include <chrono>

using namespace std;

void SelectionSort(int A[],int tam){
    for (int i = 0; i < tam; i++) {
        int menor = i;
        for (int j = i + 1; j < tam; j++) {
            if (A[j] < A[menor]) {
            	//Encontrar el N°menor
                menor = j;     
            }
        }
        //intercambiamos con la posicion i el N°menor
        int temp = A[i];
        A[i] = A[menor];
        A[menor] = temp;
    }	
}

void printArray(int a[], int n) {
  for (int i = 0; i < n; ++i) {
    printf("%d ", a[i]);
  }
  printf("\n");
}

int main() {
  int n = 0;
  while (std::cin >> n) {
    int arr[n] = {0};
    for (int i = 0; i < n; ++i) {
      std::cin >> *(arr + i);
    }
    auto start = std::chrono::steady_clock::now();
    SelectionSort(arr, n);
    auto end = std::chrono::steady_clock::now();
    // printf("Sorted array: \n");
    // printArray(arr, n);
    std::chrono::duration<double> elapsed_seconds = end - start;
    std::cout << elapsed_seconds.count() << ' ';
  }
  std::cout << '\n';

  return 0;
}
